pkg/sink/sink_test.go

```diff
@@ -359,28 +349,259 @@ func Test_HandleEvent(t *testing.T) {
+						return true, obj, nil
+					})
+				return pClient
+			}(),
+			wantErr: true,
+		},
+		{
+			name: "Test_SecureEndpoint_TaskrunCreateFailure",
+			pClient: func() *fakepipelineclientset.Clientset {
+				ctx, _ := rtesting.SetupFakeContext(t)
+				clients, _ := pipelinetest.SeedTestData(t, ctx, pipelinetest.Data{
+					Tasks: []*pipelinev1.Task{task},
+				})
+				pClient := clients.Pipeline
+				pClient.PrependReactor("create", "taskruns",
+					func(action k8stest.Action) (bool, runtime.Object, error) {
+						return true, nil, errors.New("mock create taskrun error")
+					})
+				return pClient
+			}(),
+			wantErr: true,
+		},
+		{
+			name: "Test_SecureEndpoint_TaskrunGetFailure",
+			pClient: func() *fakepipelineclientset.Clientset {
+				ctx, _ := rtesting.SetupFakeContext(t)
+				clients, _ := pipelinetest.SeedTestData(t, ctx, pipelinetest.Data{
+					Tasks: []*pipelinev1.Task{task},
+				})
+				pClient := clients.Pipeline
+				pClient.PrependReactor("get", "taskruns",
+					func(action k8stest.Action) (bool, runtime.Object, error) {
+						return true, nil, errors.New("mock create taskrun error")
+					})
+				return pClient
+			}(),
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			r.PipelineClient = tt.pClient
+			if err := r.validateEvent(triggerValidate, h, payload); (err != nil) != tt.wantErr {
+				t.Errorf("Resource.secureEndpoint() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func TestResource_createValidateTask(t *testing.T) {
+	r := Resource{
+		EventListenerName:      "foo-listener",
+		EventListenerNamespace: "foo",
+	}
+
+	h := http.Header{}
+	h.Set("X-Hub-Signature", "1234567")
+	h["X-Hub-Array"] = []string{"1234567", "abcde"}
+
+	payload := []byte("test payload")
+	triggerValidate := &triggersv1.TriggerValidate{
+		TaskRef: pipelinev1.TaskRef{
+			Name: "bar",
+		},
+		ServiceAccountName: "foo",
+		Params: []pipelinev1.Param{{
+			Name: "Secret",
+			Value: pipelinev1.ArrayOrString{
+				Type:      pipelinev1.ParamTypeString,
+				StringVal: "github-secret",
+			},
+		}},
+	}
+
+	tests := []struct {
+		name    string
+		pClient *fakepipelineclientset.Clientset
+		want    *pipelinev1.TaskRun
+		wantErr bool
+	}{
+		{
+			name: "Test_createValidateTask",
+			pClient: func() *fakepipelineclientset.Clientset {
+				ctx, _ := rtesting.SetupFakeContext(t)
+				clients, _ := pipelinetest.SeedTestData(t, ctx, pipelinetest.Data{
+					Tasks: []*pipelinev1.Task{pipelinetb.Task("bar", "foo", pipelinetb.TaskSpec(
+						pipelinetb.TaskInputs(
+							pipelinetb.InputsParamSpec("X-Hub-Signature", v1alpha1.ParamTypeString, pipelinetb.ParamSpecDescription("mydesc"), pipelinetb.ParamSpecDefault("default")),
+							pipelinetb.InputsParamSpec("X-Hub-Array", v1alpha1.ParamTypeArray, pipelinetb.ParamSpecDescription("desc"), pipelinetb.ParamSpecDefault("array", "values")))))},
+				})
+				return clients.Pipeline
+			}(),
+			want: &pipelinev1.TaskRun{
```

Comment:
I think it's fine to using test builders here as well. They have already been tested themselves as producing the same output as stubs.